docker/local-build.sh

```diff
@@ -39,19 +42,19 @@ echo "$BUILD_NUMBER" > $WORKSPACE/.build_number
 CI_BUILD_ID="${BUILD_TIMESTAMP}-build-${BUILD_NUMBER}"
 
 # export other vars
-for var in WORKSPACE BASE_DISTRO CURRENT_PROJECT BUILD_CACHE_DIR GIT_PROXY_COMMAND CI_BUILD_ID TARGET_MACHINE; do
-	RUN_ARGS="$RUN_ARGS -e $var=${!var}"
+for var in WORKSPACE BASE_DISTRO CURRENT_PROJECT BUILD_CACHE_DIR GIT_PROXY_COMMAND CI_BUILD_ID TARGET_MACHINE GIT_COMMITTER_NAME GIT_COMMITTER_EMAIL; do
+	RUN_ARGS+=(-e "$var=${!var}")
 done
 # Point HOME to WORKSPACE, don't polute real home.
-RUN_ARGS="$RUN_ARGS -e HOME=$WORKSPACE"
+RUN_ARGS+=(-e "HOME=$WORKSPACE")
 
 docker build -t $CURRENT_PROJECT $BUILD_ARGS $WORKSPACE/docker/$BUILDOS
 
 if [ ! -d $BUILD_CACHE_DIR ]; then
     mkdir -p $BUILD_CACHE_DIR
 fi
 
-docker run -it --rm $RUN_ARGS \
+docker run -it --rm "${RUN_ARGS[@]}" \
```

Comment:
GIT_* variables I don't think it's really good idea to pass. Same with *_proxy variables: better to sanitize  content of *_proxy variables.
However, if you split patch for making RUN_ARGS as array and test it under different versions of bash (3 and 4 at least), I think it can be accepted.
Another potential enhancement that can be done in build scripts is to parse BB_ENV_EXTRAWHITE and pass those as arguments to docker calls. that way user would have explicit way to pass something from his development environment into isolated container.